libsolidity/AST.cpp

```diff
@@ -403,7 +403,8 @@ void Assignment::checkTypeRequirements()
 	m_leftHandSide->checkTypeRequirements();
 	m_leftHandSide->requireLValue();
 	//@todo later, assignments to structs might be possible, but not to mappings
-	if (!m_leftHandSide->getType()->isValueType() && !m_leftHandSide->isLocalLValue())
+	if (m_leftHandSide->getType()->getCategory() != Type::Category::ByteArray &&
+			!m_leftHandSide->getType()->isValueType() && !m_leftHandSide->isLocalLValue())
```

Comment:
Yes, we should, but actually only mappings are not assignable.
For enums, you should not need to edit this, since enum should be a value type (it is always passed by value).